tests/test_utils/test_nan_warnings.py

```diff
@@ -0,0 +1,145 @@
+"""Test the RadViz feature analysis visualizers."""
+
+import numpy as np
+import pytest
+
+from yellowbrick.exceptions import DataWarning
+from yellowbrick.utils.nan_warnings import count_nan_elements, \
+    count_rows_with_nans, warn_if_nans_exist, clean_data
+
+
+def test_raise_warning_if_nans_exist():
+    """Test that a warning is raised if any nans are in the data."""
+    data = np.array([
+        [1, 2, 3],
+        [1, 2, np.nan],
+    ])
+
+    with pytest.warns(DataWarning):
+        warn_if_nans_exist(data)
+
+
+def test_count_rows_in_2d_arrays_with_nans():
+    """Test that a warning is raised if any nans are in the data."""
+    data0 = np.array([
+        [1, 2, 3],
+    ])
+
+    data2 = np.array([
+        [1, 2, 3],
+        [1, 2, 3],
+        [np.nan, 2, 3],
+        [1, np.nan, 3],
+    ])
+
+    data3 = np.array([
+        [1, 2, 3],
+        [np.nan, 2, 3],
+        [1, np.nan, 3],
+        [np.nan, np.nan, np.nan],
+    ])
+
+    assert count_rows_with_nans(data0) == 0
+    assert count_rows_with_nans(data2) == 2
+    assert count_rows_with_nans(data3) == 3
+
+
+def test_count_nan_elements():
+    """Test that a warning is raised if any nans are in the data."""
+    data0 = np.array([1, 2, 3])
+    data1 = np.array([1, np.nan, 3])
+    data3 = np.array([np.nan, np.nan, np.nan])
+
+    assert count_nan_elements(data0) == 0
+    assert count_nan_elements(data1) == 1
+    assert count_nan_elements(data3) == 3
+
+
+def test_clean_data_X_only_no_nans():
+    """Test that an array with no nulls is returned intact."""
+    X = np.array([
+        [1, 2, 3],
+        [4, 5, 6],
+        [7, 8, 9],
+    ])
+
+    observed = clean_data(X)
+    np.testing.assert_array_equal(X, observed)
+
+
+def test_clean_data_X_only():
+    """Test that arrays with missing data are returned without nans."""
```

Comment:
Same as above for these docstrings, though they do actually describe what's happening!